app/models/user.rb

```diff
@@ -0,0 +1,34 @@
+require 'bcrypt'
+
+class User
+  include DataMapper::Resource
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  property :id, Serial
+  property :full_name, String
+  property :email, String, unique: true, message: 'This email is already taken'
+  property :username, String, unique: true, message: 'This username is already taken'
+
+  property :password_digest, Text
+
+  def password=(password)
+    @password = password
+    self.password_digest = BCrypt::Password.create(password)
+  end
+
+  validates_confirmation_of :password
```

Comment:
Prefer the new style validations `validates :column, confirmation: value` over `validates_confirmation_of`.